### PR TITLE
DOC: optimize.linprog: adjust HiGHS URL

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Per request of the HiGHS team

#### What does this implement/fix?
Change references to https://www.maths.ed.ac.uk/hall/HiGHS to https://highs.dev/

### CODE REVIEW

File: scipy/optimize/_highs/cython/src/_highs_wrapper.pyx [hunk 1]
```diff
@@ -555,7 +555,7 @@ def _highs_wrapper(
 
     References
     ----------
-    .. [1] https://www.maths.ed.ac.uk/hall/HiGHS
+    .. [1] https://highs.dev/
     .. [2] https://www.maths.ed.ac.uk/hall/HiGHS/HighsOptions.html
```
Comment (mdhaber) [inline scipy/optimize/_highs/cython/src/_highs_wrapper.pyx:559]: @jajhall I wanted to replace this link with the equivalent at highs.dev but all I could find was:
https://ergo-code.github.io/HiGHS/man/options.html
which is not really a substitute. Is it OK to leave this link to https://www.maths.ed.ac.uk/hall/HiGHS/HighsOptions.html or is there a better version at https://highs.dev/?

Comment (jajhall) [reply scipy/optimize/_highs/cython/src/_highs_wrapper.pyx:559]: There's no better version: our options aren't currently documented, I'm afraid. We were a bit keen to delete things from https://highs.dev/, expecting [https://ergo-code.github.io/HiGHS](https://ergo-code.github.io/HiGHS) to be populated much more rapidly.

Comment (mdhaber) [reply scipy/optimize/_highs/cython/src/_highs_wrapper.pyx:559]: It only appears in this one private file, so I think I'll leave it alone. Thanks!

**Review by tylerjereddy — APPROVED** [review]
Simple doc link modernization and doc build is passing.

**Pull request merged**